src/Query/Grouper.ts

```diff
@@ -1,12 +1,37 @@
 import type { Task } from '../Task';
 
 /**
- * A naming function, that takes a Task object and returns the corresponding group property name
+ * A group-naming function, that takes a Task object and returns zero or more
+ * corresponding group property names.
+ *
+ * For example, if a {@link GrouperFunction} implemented grouping by tag,
+ * passing in a task that has two tags, #tag1 and #tag2 to this tag-grouper function
+ * would return ['tag1', 'tag2']
  */
 export type GrouperFunction = (task: Task) => string[];
 
+/**
+ * A named function that is used to determine the group heading name(s) to use for a {@link Task} object.
+ *
+ * The name is represented in {@link property}.
+ *
+ * Note: {@link Grouper} objects are typically created by {@link Field.grouper} using the many
+ * classes derived from {@link Field}.
+ *
+ * @see {@link TaskGroups} for how to use {@link Grouper} objects to group tasks together.
+ */
 export class Grouper {
+    /**
+     * The type of grouper, for example 'tags' or 'due'.
```

Comment:
I don't believe we have types for groupers. Actually this `property is used ONLY for CSS styles (Through `QueryRenderer.getGroupingAttribute()`).

And see `Field` and `MultiTextField`, it contains the field name to be grouped upon. Maybe that is the correct thing to reference in the comment & the variable name?

Reply:
> I don't believe we have types for groupers. Actually this `property is used ONLY for CSS styles (Through `QueryRenderer.getGroupingAttribute()`).

When we talked about that yesterday I realised that we haven't finished migrating the grouping fully to the Field hierarchy.

Which is why this is so tacky:

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/980b068f61c64bb0aa08e96e9f15992ff0521950/src/Query/FilterParser.ts#L93-L121

Reply:
ok, great point, maybe a topic for a future pairing session =)